packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx

```diff
@@ -65,31 +59,30 @@ export function CrossmintAuthProvider({
     loginMethods = ["email", "google"],
 }: CrossmintAuthProviderProps) {
     const [user, setUser] = useState<SDKExternalUser | undefined>(undefined);
-    const { crossmint, setJwt, setRefreshToken } = useCrossmint(
-        "CrossmintAuthProvider must be used within CrossmintProvider"
-    );
-    const crossmintAuthService = new CrossmintAuthService(crossmint.apiKey);
+    const { crossmint, setJwt } = useCrossmint("CrossmintAuthProvider must be used within CrossmintProvider");
+    // Only create the CrossmintAuth instance once, even in StrictMode, as the constructor calls /refresh
+    // It can only be called once to avoid race conditions
+    const crossmintAuthRef = useRef<CrossmintAuth | null>(null);
+    // biome-ignore lint/correctness/useExhaustiveDependencies: crossmint can't be a dependency because it updates with each jwt change
+    const crossmintAuth = useMemo(() => {
+        if (!crossmintAuthRef.current) {
+            crossmintAuthRef.current = CrossmintAuth.from(crossmint, {
```

Comment:
Ooooh now I understand the need for hooks. I thought it was for the developer, but it's actually to let us hook here in react.
I wonder if that could also confuse developers in the future and they won't know what to do on their hooks.

Reply:
not sure what you're referring to

Reply:
the `onLogout` and `onTokenRefresh` "hooks" are defined in the auth-sdk-client and then used here.
What's the expected usage of those methods if used outside of the `react-ui` sdk?

Reply:
Other apps that use the auth client sdk directly, instead of the react one, will want to do similar actions as the ones we take here in the react sdk. The idea of the client auth sdk is to be framework agnostic

Reply:
Great! Now I understand the separation:
- CrossmintAuthClient will remove the cookies (cross-framework way of storing the material)
- React-UI will hook to handle component state change
  - Other frameworks might want to do the same
  - When using React-UI, developers are not expected to add their own hooks to `onLogout` or `onRefreshToken`.